File: app_java/src/main/java/vlab/android/architecture/feature/login/LoginErrorHandler.java
package vlab.android.architecture.feature.login;

import vlab.android.architecture.R;
import vlab.android.architecture.base.BaseErrorHandler;

/**
 * Created by Vinh Tran on 3/3/18.
 */

public class LoginErrorHandler extends BaseErrorHandler {

    @Override
    public String parseError(Throwable throwable){
        if(throwable == null){
            return mContext.getString(R.string.login_failed);
        }
        return mContext.getString(R.string.login_failed) + " : " + throwable.getMessage();
    }
}

File: app_kotlin/src/main/java/vlab/android/architecture/feature/login/LoginErrorHandler.kt
package vlab.android.architecture.feature.login

import vlab.android.architecture.R
import vlab.android.architecture.base.BaseErrorHandler

/**
 * Created by Vinh Tran on 3/3/18.
 */

class LoginErrorHandler : BaseErrorHandler() {

    override fun parseError(throwable: Throwable): String {
        return if (throwable == null) {
            mContext.getString(R.string.login_failed)
        } else {
            mContext.getString(R.string.login_failed) + " : " + throwable.message
        }
    }
}
